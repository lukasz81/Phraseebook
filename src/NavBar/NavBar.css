nav {
  background-color: #FE5000;
  min-height: 64px;
  color: #FFFFFF; }
