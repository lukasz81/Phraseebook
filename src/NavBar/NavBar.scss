@import "../variables.scss";

@media only screen and (min-width: $break-main) {
  .notification-wrapper {
    background: #0000000f
  }
  .notifications-count {
    border: 2px solid #0000000f;
  }
}

nav {
  background-color: $main-color;
  min-height: $navSize;
  color: $nav-text-color;
  svg {
    margin: auto;
  }
  .notification-wrapper {
    width: 100%;
    cursor: pointer;
  }
  .notifications-count {
    display: block;
    width: 20px;
    height: 20px;
    background: white;
    color: #fe5000;
    text-align: center;
    border-radius: 50px;
    position: absolute;
    right: 5px;
    top: 5px;
    line-height: 20px;
    font-weight: 600;
    border: 2px solid;
    font-size: 12px;
  }
  .notification-actions {
    width: $navSize;
    height: $navSize;
    position: relative;
    cursor: pointer;
    transition: all 0.2s ease-in-out;
    justify-content: space-between;
    &:hover, &.active {
      background: white;
      svg {
        fill: $main-color
      }
      .notifications-count {
        color: white;
        background: $main-color;
      }
    }
  }
}