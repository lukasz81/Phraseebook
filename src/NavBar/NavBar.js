import React from 'react';
import './NavBar.css';
import { MdNotifications } from 'react-icons/md';
import {withRouter} from 'react-router-dom';

const NavBar = (props) => {

    const notificationsPath = props.location.pathname === '/notifications';

    const onClickHandler = () => {
        props.history.push('/notifications')
    };

    const goHomeHandler = () => {
        props.history.push('/')
    };

    return (
        <nav className={'flex-container'}>
            <div className={'flex-container flex-row notification-wrapper'}>
                <div onClick={() => goHomeHandler()} className={'default-padding button-1'}>
                    <b>Phrasebook</b>
                </div>
                <div className={notificationsPath ? 'notification-actions flex-container active' : 'notification-actions flex-container' } onClick={() => onClickHandler()}>
                    <MdNotifications size={30}/>
                    {(!props.props.isNotificationRead && props.props.notifications) && <span className={'notifications-count'}>{props.props.notifications ? props.props.notifications.length : '...'}</span>}
                </div>
            </div>
        </nav>
    );
};

export default withRouter(NavBar);