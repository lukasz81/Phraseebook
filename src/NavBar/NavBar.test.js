import NavBar from './NavBar';
import Enzyme, {shallow} from "enzyme";
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

describe('Tests button component',() => {

    let historyMock;
    let mock = jest.fn();
    historyMock = { push: mock };

    const makeComponent = (isNotificationRead = false) => {
        return shallow(
            <NavBar.WrappedComponent
                location={{pathname: '/'}}
                history={historyMock}
                props={{props:{
                        isNotificationRead: isNotificationRead,
                        notifications: ['6']
                    }}}
            />
        )
    };

    afterEach(() => {
        mock.mockRestore();
    });

    it('should render correctly',() => {
        expect(makeComponent()).toMatchSnapshot();
    });

    it('should update history location to "/"',() => {
        const component = makeComponent();
        const element = component.find('.button-1');
        element.simulate('click');
        expect(historyMock.push.mock.calls[0]).toEqual([ ('/') ]);
    });

    it('should update history location to "/notifications"',() => {
        const component = makeComponent();
        const element = component.find('.notification-actions');
        element.simulate('click');
        expect(historyMock.push.mock.calls[0]).toEqual([ ('/notifications') ]);
    });

    it('should not show count if marked as read',() => {
        const component = makeComponent(true);
        const element = component.find('.notifications-count');
        expect(element.length).toEqual(0);
    });

});
