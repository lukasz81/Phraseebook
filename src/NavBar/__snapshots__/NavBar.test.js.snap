// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests button component should render correctly 1`] = `
<nav
  className="flex-container"
>
  <div
    className="flex-container flex-row notification-wrapper"
  >
    <div
      className="default-padding button-1"
      onClick={[Function]}
    >
      <b>
        Phrasebook
      </b>
    </div>
    <div
      className="notification-actions flex-container"
      onClick={[Function]}
    >
      <MdNotifications
        size={30}
      />
    </div>
  </div>
</nav>
`;
